src/mono/mono/mini/llvmonly-runtime.c

```diff
@@ -215,6 +215,29 @@ mini_llvmonly_add_method_wrappers (MonoMethod *m, gpointer compiled_method, gboo
 	return addr;
 }
 
+static inline MonoVTableEEData*
+get_vtable_ee_data (MonoVTable *vtable)
+{
+	MonoVTableEEData *ee_data = (MonoVTableEEData*)vtable->ee_data;
+
+	if (G_UNLIKELY (!ee_data)) {
+		ee_data = m_class_alloc0 (vtable->klass, sizeof (MonoVTableEEData));
+		mono_memory_barrier ();
+		vtable->ee_data = ee_data;
+	}
+	return ee_data;
```

Comment:
Similar to what @BrzVlad said: isn't there still a race condition here?  Two threads can both call `m_class_alloc0` and both return their own freshly-allocated ee_data.  

Also, you have 2 copies of this function. here and also in `interp.c`
```suggestion
	return ee_data;
```
```suggestion
	return ee_data;
```

Reply:
Will fix that in a follow up PR.